pyramid_openapi3/tests/test_validation.py

```diff
@@ -0,0 +1,198 @@
+        view = self.config.registry.adapters.registered(
+            (IViewClassifier, request_interface, Interface), IView, name=""
+        )
+        return view
+
+    def _get_request(self, params: t.Dict = None) -> DummyRequest:
+        """Create a DummyRequest instance matching example view.
+
+        :param params: Query parameter dictionary
+        """
+        request = DummyRequest(config=self.config, params=params)
+        request.matched_route = DummyRoute(name="foo", pattern="/foo")
+        return request
+
+    def test_request_validation_error(self) -> None:
+        """Test View raises RequestValidationError.
+
+        Request parameters don't match schema.
+        """
+        self._add_view()
+        view = self._get_view()
+        request = self._get_request()
+        from pyramid_openapi3 import RequestValidationError
+
+        with self.assertRaises(RequestValidationError) as cm:
+            view(None, request)
+        response = cm.exception
+        self.assertEqual(str(cm.exception), "Missing required parameter: bar")
+        # not enough of pyramid has been set up so we need to render the
+        # exception response ourselves.
+        response.prepare({"HTTP_ACCEPT": "application/json"})
+        self.assertIn("Missing required parameter: bar", response.json["message"])
+
+    def test_response_validation_error(self) -> None:
+        """Test View raises ResponseValidationError.
+
+        Example view raises an undefined response code.
+        """
+        from pyramid.httpexceptions import HTTPPreconditionFailed
+
+        self._add_view(lambda *arg: (_ for _ in ()).throw(HTTPPreconditionFailed()))
+        view = self._get_view()
+        request = self._get_request(params={"bar": "1"})
+        from pyramid_openapi3 import ResponseValidationError
+
+        with self.assertRaises(ResponseValidationError) as cm:
+            view(None, request)
+        response = cm.exception
+        # not enough of pyramid has been set up so we need to render the
+        # exception response ourselves.
+        response.prepare({"HTTP_ACCEPT": "application/json"})
+        self.assertIn("Unknown response http status: 412", response.json["message"])
+
+    def test_view_raises_valid_http_exception(self) -> None:
+        """Test View raises HTTPException.
+
+        Example view raises a defined response code.
+        """
+        from pyramid.httpexceptions import HTTPBadRequest
+
+        self._add_view(lambda *arg: (_ for _ in ()).throw(HTTPBadRequest("failed")))
+        view = self._get_view()
+        request = self._get_request(params={"bar": "1"})
+        with self.assertRaises(HTTPBadRequest) as cm:
+            view(None, request)
+        response = cm.exception
+        # not enough of pyramid has been set up so we need to render the
+        # exception response ourselves.
+        response.prepare({"HTTP_ACCEPT": "application/json"})
+        self.assertIn("failed", response.json["message"])
```

Comment:
`failed` is a too generic word. It could be raised by some other error, and this test would think everything is good. Please use a string that is less likely to be present in arbitrary error messages, for example `foo` or `foobar` or `this is a bad request`.

Reply:
Yes, I was thinking the same when I wrote that test :). Will update the pull request here.

Reply:
fixed